README.md

```diff
@@ -1,10 +1,129 @@
 # GenZ
-## Requirements
 
-- [Go >1.20](https://go.dev/doc/install)
-## Try it out
-Explore built-in `examples`, clone repo, and run
+[![Go Reference](https://pkg.go.dev/badge/github.com/Joffref/genz.svg)](https://pkg.go.dev/github.com/Joffref/genz)
+[![Go Report Card](https://goreportcard.com/badge/github.com/Joffref/genz)](https://goreportcard.com/report/github.com/Joffref/genz)
+[![License](https://img.shields.io/github/license/Joffref/genz)](LICENSE)
+![GitHub stars](https://img.shields.io/github/stars/Joffref/genz)
+
+> At this moment, this project is in development. It is not ready for production use. See the [Contributing](#contributing) section if you would like to help make it better.
+
+✨ Never write toil/boilerplate code again! ✨
+
+Thanks to GenZ, you can generate code for your own needs, without having to write a single line of code.
+
+<p align="center" heigth="50%">
+    <img heigth="50%" src="./docs/assets/logo_500_500.png"> 
+</p>
+
+GenZ is a template-based generator for Go. A single binary that can be called with the native `go generate` to automate generation of your Go code based on templates.
+
+## Installation
+
+You can install GenZ in different ways, depending on your needs.
+We strongly recommend using the Go version for production use.
+
+### Using Go - Production
+
+```bash
+go install github.com/Joffref/genz
+```
+
+### From source - Development
+
+```bash
+git clone https://github.com/Joffref/genz.git
+cd genz
+make install
+```
+
+## Getting Started
+
+Let's say you have several types defined in your project (such as `Human` as defined below).
+```go
+package main
+
+type Human struct {
+	// +required
+	Firstname string
+	// +required
+	Lastname string
+	Age uint
+}
+```
+And you want to add a `Validate()` method on these type.
+
+With GenZ, all you have to do is to write a template once:
+```gotemplate
+package main
+
+import (
+    "fmt"
+)
+
+func (v {{ .Type }}) Validate() error {
+{{ range .Attributes }} {{$attribute := .}}
+{{ if eq .Type "string" }}{{ range .Comments }}
+{{ if eq . "+required" }}
```

Comment:
Example won't work, comment is equal to " +required" (with a space) in the example above